transcode/transcode.go

```diff
@@ -109,9 +110,20 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 	// transcodedStats hold actual info from transcoded results within requested constraints (this usually differs from requested profiles)
 	transcodedStats := statsFromProfiles(transcodeProfiles)
 
+	renditionList := video.TRenditionList{RenditionSegmentTable: make(map[string]video.TSegmentList)}
+	// only populate video.TRenditionList map if MP4 is enabled via override or short-form video detection
+	if transcodeRequest.GenerateMP4 {
+		for _, profile := range transcodeProfiles {
+			renditionList.AddRenditionSegment(profile.Name,
+				video.TSegmentList{
+					SegmentDataTable: make(map[int][]byte),
+				})
+		}
+	}
```

Comment:
it could be nice to just late initialise these inside `video/media.go` when calls are made to `GetSegmentList` further down